src/games/strategy/triplea/delegate/StrategicBombingRaidBattle.java

```diff
@@ -203,7 +203,16 @@ public void fight(final IDelegateBridge bridge) {
     showBattle(bridge);
     final List<IExecutable> steps = new ArrayList<>();
     if (hasAA) {
-      steps.add(new FireAA());
+    	boolean foundGenericAA = false;
+      for (final Unit target : m_targets.keySet()) {
+    		//for( Unit attacker
+    		if( ( (UnitAttachment) target.getType().getAttachment("unitAttachment")).getIsAAforBombingThisUnitOnly() ) {
+    			final Collection<Unit> currentAttackers = m_targets.get(target);
+    			if( currentAttackers != null ) steps.add(new FireAA(currentAttackers));
+    		}
+    		else foundGenericAA = true;
+    	}
+      if( foundGenericAA ) steps.add(new FireAA());
```

Comment:
Please look to the style code, the auto-formatter has not been run. There should always be open and close braces on all if and else statements. This code should at least look similar to the rest of the class and the rest of the package.